controllers/helloworld_controller.go

```diff
@@ -0,0 +1,201 @@
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/example/memcached-operator/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// HelloworldReconciler reconciles a Helloworld object
+type HelloworldReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=cache.my.domain,resources=helloworlds,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=cache.my.domain,resources=helloworlds/status,verbs=get;update;patch;create
+//+kubebuilder:rbac:groups=cache.my.domain,resources=helloworlds/finalizers,verbs=update
+
+//+kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=services,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Helloworld object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *HelloworldReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	Logging(req, "Starting reconcile")
+
+	helloworld := &v1alpha1.Helloworld{}
+
+	if err := r.Get(ctx, req.NamespacedName, helloworld); err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		log.Log.Error(err, "unable to fetch Helloworld")
+		return ctrl.Result{}, err
+	}
+
+	svc := &v1.Service{}
+	svcNamespacedName := types.NamespacedName{Namespace: helloworld.Namespace, Name: helloworld.Name + "-service"}
+
+	if err := r.Get(ctx, svcNamespacedName, svc); err != nil {
+		if errors.IsNotFound(err) {
+			Logging(req, "Creating a service")
+			r.Create(ctx, r.svcForHelloworld(helloworld))
+			return ctrl.Result{Requeue: true}, nil
+		}
+		log.Log.Error(err, "unable to fetch "+helloworld.Name+"'s service")
+		return ctrl.Result{}, err
+	}
+
+	Logging(req, "Service exist - "+svc.Name)
+
+	requiredReplicas := helloworld.Spec.ReplicaCount
+
+	Logging(req, "Required count of pods is "+fmt.Sprint(requiredReplicas))
+
+	podList := &v1.PodList{}
+	opts := []client.ListOption{
+		client.InNamespace(helloworld.Namespace),
+		client.MatchingLabels{"hw_name": helloworld.Name},
```

Comment:
you should extract this string to a const and use it everywhere

Reply:
you didn't fix this

Reply:
I saw you added a function for this, but you haven't extracted the string to a const,
so you are still using the same string in 2 different places without const..